src/main/java/edu/kis/vh/nursery/HanoiRyhmer.java

```diff
@@ -1,17 +1,27 @@
 package edu.kis.vh.nursery;
 
-public class HanoiRyhmer extends defaultCountingOutRyhmer {
+public class HanoiRyhmer extends DefaultCountingOutRyhmer {
 
-int totalRejected = 0;
+    private int totalRejected = 0; // dostep do zmiennej odbywa sie poprzez funkcje get i set
 
-	public int reportRejected() {
-		return totalRejected;
-	}
+    int reportRejected() {
+        return getTotalRejected();
+    }
 
-	public void countIn(int in) {
-	if (!callCheck() && in > peekaboo())
-			totalRejected++;
-			else
-				super.countIn(in);
-	}
+    @Override
+    public void countIn(final int in) {
+        if (!callCheck() && in > peekaboo())
+            setTotalRejected(getTotalRejected() + 1);

```

Comment:
This is an exaggeration. Using private fields internally usually do not require encapsulation.